api/src/main/java/jakarta/servlet/http/Cookie.java

```diff
@@ -422,4 +423,103 @@ public void setHttpOnly(boolean isHttpOnly) {
     public boolean isHttpOnly() {
         return isHttpOnly;
     }
+
+    /**
+     * Sets the <i>SameSite</i> flag for this cookie with the provided value.
+     *
+     * <p>
+     *     When <code>sameSiteMode</code> is not <code>null</code>, this cookie will have its <code>SameSite</code>
+     *     attribute set with the provided value.
+     * </p>
+     *
+     * @param sameSiteMode The <i>SameSite</i> attribute's value. May be null if the attribute should not be set.
+     *
+     * @since Servlet 4.x
+     */
+    public void setSameSiteMode(SameSiteMode sameSiteMode) {
+        this.sameSiteMode = sameSiteMode;
+    }
+
+    /**
+     * Returns the <i>SameSite</i>> mode associated with this cookie.
+     *
+     * @return The <i>SameSite</i> mode associated with this cookie. May be null.
+     *
+     * @since Servlet 4.x
+     */
+    public SameSiteMode getSameSiteMode() {
+        return sameSiteMode;
+    }
+
+    /**
+     * The available values that can be used with the cookie's <i>SameSite</i> attribute.
+     */
+    public enum SameSiteMode {
+
+        /**
+         * This mode effectively disables the protection provided by SameSite cookies.
+         *
+         * <p>
+         *     When SameSiteMode is {@literal None}, applications should have additional measures to protect cookies
+         *     from CSRF attacks.
+         * </p>
+         *
+         * <p>
+         *     Browsers may default to {@link SameSiteMode#LAX} or {@link SameSiteMode#STRICT} when the flag is not
+         *     explicitly set by applications in the Set-Cookie header. Setting the {@literal SameSite} flag to
+         *     {@literal None} ensures that this default behavior is disabled in browsers.
+         * </p>
+         *
+         * <p>
+         *     Note that this mode is browser dependent, and browsers may require cookies to have the {@literal Secure}
+         *     flag to be set at the same time.
+         * </p>
+         */
+        NONE("None"),
```

Comment:
I don't think we need the mixed case name for the enum.   Sure it looks nicer sometimes, but all other enums are just the upper case version.

Reply:
My intention here was probably to make app servers' job a bit easier when it comes to constructing the Set-Cookie headers. If this is not intended, I'll drop these.

Reply:
Hmmm torn.   It would be nice to not have to have implementations do the conversion to the cased version requred by the RFC.  So let's leave as you have it for now.